### PR TITLE
added sensors for Left Right up and down

### PR DESCRIPTION
**Pull request by amantham20**
for the we can see how fan each side is.

<sub> Mention [stepsize] in a comment if you'd like to report some technical debt. See examples [here](https://app.stepsize.com/api/demo-pr-redirect).</sub>

### CODE REVIEW

File: source/Group7_GP_Agent/GPAgentSensors.hpp [hunk 1]
```diff
@@ -0,0 +1,85 @@
+
+#pragma once
+
+#include "source/core/WorldGrid.hpp"
+
+const int WALL = 2;
```
Comment (ssitu) [inline source/Group7_GP_Agent/GPAgentSensors.hpp]: We will have to discuss with the world groups to see if they can expose the IDs of obstacles if we want this to be robust.

File: source/core/Logging.hpp
Comment (ssitu) [inline source/core/Logging.hpp:1]: hmmmmm....

**Review by ssitu — CHANGES_REQUESTED** [review]
Needs more unit tests 😅

File: source/Group7_GP_Agent/GPAgentSensors.hpp (hunk 1 above)
Comment (amantham20) [reply source/Group7_GP_Agent/GPAgentSensors.hpp]: Fair enough. 

I was also wondering about that if we can have a set of non penetrable object id's

Comment (amantham20) [reply source/Group7_GP_Agent/GPAgentSensors.hpp]: @ssitu at this moment I can add a `todo` there are we can come back to it. I started a thread on Discord.

File: source/Group7_GP_Agent/agent.hpp [hunk 2]
```diff
@@ -59,13 +61,25 @@ namespace cowboys
                             const cse491::item_set_t &item_set,
                             const cse491::agent_set_t &agent_set) override
         {
+
+            Sensors::wallDistance(grid, *this, SensorDirection::LEFT);
+            Sensors::wallDistance(grid, *this, SensorDirection::RIGHT);
+            Sensors::wallDistance(grid, *this, SensorDirection::ABOVE);
+            Sensors::wallDistance(grid, *this, SensorDirection::BELOW);
```
Comment (ssitu) [inline source/Group7_GP_Agent/agent.hpp:70]: What're these doing here?

**Review by ssitu — COMMENTED** [review]
There's discarded calls to wallDistance that could be removed.

**Review by ssitu — APPROVED** [review]
Looks good

**Review by Rajmeet — APPROVED** [review]
looks a good

**Pull request merged**